src/events/http/HttpServer.js

```diff
@@ -412,6 +412,39 @@ export default class HttpServer {
     return authStrategyName
   }
 
+  _setAuthorizationStrategy(endpoint, functionKey, method, path) {
+    /*
+     *  The authentication strategy can be provided outside of this project
+     *  by injecting the provider through a custom variable in the serverless.yml.
+     *
+     *  see the example in the tests for more details
+     *    /tests/integration/custom-authentication
+     */
+    const customizations = this.#serverless.service.custom
+    if (
+      customizations &&
+      customizations.offline?.customAuthenticationProvider
+    ) {
+      // eslint-disable-next-line import/no-dynamic-require, global-require
+      const provider = require(customizations.offline
```

Comment:
This `require` will search in context of a directory in which this `HttpServer.js` file is placed. While ideally we should search against _service directory_ (it's also how we load service local plugins).

For that you can rely on https://nodejs.org/api/module.html#modulecreaterequirefilename, as we do indirectly in Framework here: https://github.com/serverless/serverless/blob/480d74ec58c78466d2d055e5c2d73c2f4dde0b42/lib/classes/plugin-manager.js#L159

Reply:
@medikoo - thanks for the feedback.  I knew that the search would be relative to the HttpServer and wasn't thrilled with the confusion this could create but I didn't know this could be modified via createRequire.  

I think this matches your suggestions, let me know if it doesn't.